### PR TITLE
H virtual joystick

### PR DESCRIPTION
**Pull request by hema7dubal**

### CODE REVIEW

File: LogHammer/Assets/Scripts/BallMovement.cs [hunk 1]
```diff
@@ -2,13 +2,19 @@
 using System.Collections.Generic;
 using UnityEngine;
 
+//Handles the movement of the Player
 public class BallMovement : MonoBehaviour
 {
+    //to store the reference of rigidbody component of the player, to which we will apply some force, for its movement
```
Comment (parth16parikh) [inline LogHammer/Assets/Scripts/BallMovement.cs]: Lets start using standard created by you in this code

File: LogHammer/Assets/Scripts/BallMovement.cs [hunk 2]
```diff
@@ -2,13 +2,19 @@
...
     private Rigidbody rigidBody;
```
Comment (parth16parikh) [inline LogHammer/Assets/Scripts/BallMovement.cs]: Why do you need to store players rigidbody? you should just provide 2d Vector.

File: LogHammer/Assets/Scripts/BallMovement.cs [hunk 3]
```diff
@@ -23,16 +29,23 @@ static public BallMovement Instance
...
+
+        //assign rigidbody instance
         rigidBody = GetComponent<Rigidbody>();
     }
 
+    //moves the ball according to the Force
     public void MoveBall(Vector3 force)
     {
-        print(force);
+        //the input that we got has opposite axes, hence resetting the axes here
```
Comment (parth16parikh) [inline LogHammer/Assets/Scripts/BallMovement.cs]: dont use numbers like -1 directly. Use macro or some static constant

**Review by parth16parikh — CHANGES_REQUESTED** [review]
change this things and i will check all the things on your laptop

File: LogHammer/Assets/Scripts/BallMovement.cs (hunk 1 above)
Comment (hema7dubal) [reply LogHammer/Assets/Scripts/BallMovement.cs]: okay

File: LogHammer/Assets/Scripts/BallMovement.cs (hunk 2 above)
Comment (hema7dubal) [reply LogHammer/Assets/Scripts/BallMovement.cs]: 2d vector? I am using AddForce(vector3 force) method of RigidBody. That is why I took its reference here.

**Review by parth16parikh — APPROVED** [review]

**Pull request merged**